Classes/Command/NodeIndexCommandController.php

```diff
@@ -234,131 +196,377 @@ public function indexNodeCommand(string $identifier, string $workspace = null):
+    }
+
+    /**
+     * @param string $command
+     * @param array $arguments
+     * @throws Exception
+     */
+    protected function executeInternalCommand($command, array $arguments)
+    {
+        $this->outputLine();
+        $commandIdentifier = 'flowpack.elasticsearch.contentrepositoryadaptor:nodeindex:' . $command;
+        $status = Scripts::executeCommand($commandIdentifier, $this->flowSettings, true, $arguments);
+        if ($status !== true) {
+            throw new Exception(\vsprintf('Command: %s with parameters: %s', [$commandIdentifier, \json_encode($arguments)]), 1426767159);
+        }
+        $this->outputLine();
+    }
+
+    /**
+     * @param array $arguments
+     * @return int
+     * @throws Exception
+     */
+    protected function executeBuildWorkspaceCommand(array $arguments)
+    {
+        ob_start(null, 1<<20);
+        $commandIdentifier = 'flowpack.elasticsearch.contentrepositoryadaptor:nodeindex:buildworkspaceinternal';
+        $status = Scripts::executeCommand($commandIdentifier, $this->flowSettings, true, $arguments);
+        if ($status !== true) {
+            throw new Exception(\vsprintf('Command: %s with parameters: %s', [$commandIdentifier, \json_encode($arguments)]), 1426767159);
+        }
+        $output = explode(\PHP_EOL, ob_get_clean());
+        $count = (int)\array_pop($output);
+        if (count($output) > 0) {
+            foreach ($output as $line) {
+                $line = trim($line);
+                if ($line === '') {
+                    continue;
+                }
+                $this->outputLine('<info>+</info> %s', [$line]);
             }
         }
+        return $count < 1 ? 0 : $count;
+    }
+
+    /**
+     * Create a ContentContext based on the given workspace name
+     *
+     * @param string $workspaceName Name of the workspace to set for the context
+     * @param array $dimensions Optional list of dimensions and their values which should be set
+     * @return Context
+     */
+    protected function createContentContext($workspaceName, array $dimensions = [])
+    {
+        $contextProperties = [
+            'workspaceName' => $workspaceName,
+            'invisibleContentShown' => true,
+            'inaccessibleContentShown' => true
+        ];
+
+        if ($dimensions !== []) {
+            $contextProperties['dimensions'] = $dimensions;
+            $contextProperties['targetDimensions'] = array_map(function ($dimensionValues) {
+                return array_shift($dimensionValues);
+            }, $dimensions);
+        }
+
+        return $this->contextFactory->create($contextProperties);
     }
 
     /**
      * Create a new index with the given $postfix.
      *
      * @param string $postfix
+     * @param array $dimensionValues
      * @return void
-     * @throws CRAException
      */
-    protected function createNewIndex(string $postfix): void
+    protected function createNewIndex($postfix, array $dimensionValues = [])
```

Comment:
Why did you remove the $psotfix typing and the return type here?

Reply:
Because of heavy manual upmerge ... copy paste ;)